### PR TITLE
Add basic test for jsonPaymentProtocol

### PR DESCRIPTION
**Pull request by jonas-lundqvist**
Verify that signature verification works on BitPay invoices. The signature check itself is kind of Mickey Mouse-ish since the keys themselves aren't verified so this rather verifies that the payment request is well formed.

### CODE REVIEW

**Comment by cculianu** [comment]
Why are all your new wonder-tests failing?

**Comment by jonas-lundqvist** [comment]
> Why are all your new wonder-tests failing?

A stupid thing clinged on, survived and slipped thru my last refactor before I pushed it. Looking at it now I wonder why it works on my local python 3.10  environment, but that isn't something I will dwell on Friday night.

**Comment by cculianu** [comment]
Looks great to me. Thanks man!

File: electroncash/tests/test_paymentrequests.py [hunk 1]
```diff
@@ -144,3 +150,101 @@ def start_serve(self):
...
+        return response
+
+    # Verify that a payment request verification fails if the signature is incorrect
+    @mock.patch('requests.get', side_effect=mocked_bitpay_requests_get_incorrect_signature)
+    def test_bitpay_verify_incorrect_signature(self, mock_get):
+        pr = get_payment_request('https://bitpay.com/invoice')
+        self.assertTrue(pr.error is None)
+        self.assertEquals(pr.get_memo(), 'dummy_memo')
+        self.assertEquals(pr.get_payment_url(), 'https://bitpay.com/i/test123')
+        self.assertFalse(pr.verify(None))
```
Comment (cculianu) [inline electroncash/tests/test_paymentrequests.py:228]: This is broken now.  Fails here on tox. Can u fix? Run tox to see what's happening.

**Pull request merged**